control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -376,6 +376,93 @@ impl ResourcePublishing for OperationGuardArc<VolumeSpec> {
 
         self.complete_update(registry, result, spec_clone).await
     }
+
+    async fn republish(
+        &mut self,
+        registry: &Registry,
+        request: &Self::Publish,
+    ) -> Result<Self::PublishOutput, SvcError> {
+        let specs = registry.specs();
+        let state = registry.get_volume_state(&request.uuid).await?;
+
+        // If the volume is not published then it should issue publish call rather than republish.
+        if state.target.is_none() {
+            return Err(SvcError::VolumeNotPublished {
+                vol_id: state.uuid.to_string(),
+            });
+        }
+
+        let nexus_node = get_volume_target_node(registry, &state, request, true).await?;
+        let newer_nexus_id = NexusId::new();
+
+        let operation =
+            VolumeOperation::Republish((nexus_node.clone(), newer_nexus_id.clone(), request.share));
+        let spec_clone = self.start_update(registry, &state, operation).await?;
+
+        let volume_target = spec_clone.target.as_ref().expect("already validated");
+        let mut older_nexus = specs.nexus(volume_target.nexus()).await?;
+        let older_nexus_id = older_nexus.lock().clone().uuid;
```

Comment:
```suggestion
        let older_nexus_id = older_nexus.as_ref().uuid;
```

Reply:
should it be rather - > `let older_nexus_id = older_nexus.uuid().clone();` Otherwise its giving moved error.

Reply:
Right yes needs to be cloned because it's moved.